### ISSUE TITLE
Further development ideas

### ISSUE THREAD

**Issue by amichuda** [spyder-ide/spyder-reports#79]
Since version 4 is out and development of this plugin might get started again, does it still make sense to develop around `pweave` given that it's not being maintained anymore? I've had some success with `codebraid` which is actively being developed and has a similar style as `pweave`.

What do you all think and just in general, what are the plans for this plugin?

**Comment by goanpeca** [comment]
@amichuda If we can find someone that uses it and can help co maintain it then this plugin should be good to update :-)

**Comment by goanpeca** [comment]
@JamesOwers would you guys be interested in helping us co-maintain the plugin?

**Comment by JamesOwers** [comment]
Yep. However, since we last spoke I have concerns regarding the longevity of Pweave - see this convo: https://github.com/mpastell/Pweave/issues/140

How do you feel about adapting this package for use with the more active [CodeBraid](https://github.com/gpoore/codebraid)? This isn't a request to do it, as I haven't tried tried out CodeBraid yet(!), interested to know your appetite/concerns about pweave.

**Comment by goanpeca** [comment]
I am personally not using this plugin, so my input here is limited. Whatever actual people using this sort of reporting style is welcome. Since the plugin name is pretty generic, changing from one library to another is really not a issue as long as it provides the same general functionality.

If you guys agree that this is the best course of action, then let's do it ;-)

**Comment by amichuda** [comment]
That sounds awesome! I use `codebraid` for my report writing, and it works well. Having some sort of live preview would be pretty awesome. 

I don't know how much I can help with the coding here, but I can try my hand at it if I can help. I'm definitely going to test this plugin, though!

**Comment by goanpeca** [comment]
@amichuda @JamesOwers I can create issues with small snippets of code of the things that need changing. I will take a quick look at `codebraid`and see what needs to change on this plugin, but I think the needed work is very minimal.

I will try to guide you as best as I can but I would really appreciate having actual users helping develop this plugin, as you may be aware our resources are limited.


Let's start by reading this https://github.com/spyder-ide/spyder/wiki/Dev%3A-Github-Workflow

**Comment by goanpeca** [comment]
Ok so I read the Pweave thread. 

Option 1: Take over pweave
Option 2: Try codebraid.

So just to be sure, does codebraid do the things needed? OR, is it "better"to update pweave @JamesOwers ?

**Comment by amichuda** [comment]
Just to add my two cents:

I think pweave is more python specific and has a nicer syntax for code
(similar to rmarkdown), and more support for figure and table names, I
think.

Codebraid doesn't have support for figure names yet, but I think it
integrates more seamlessly with pandoc, so you can use all the same filters
that you want without switching to another package.

**Comment by goanpeca** [comment]
We could potentially offer support for both types, pweave and codebraid, but someone needs to step up, support and maintain `pweave`and the Spyder team cannot do this.

**Comment by amichuda** [comment]
That sounds like a good approach. From what I saw, there does seem to be some grumbling about someone taking over Pweave, but nothing as of yet. 

> I will try to guide you as best as I can but I would really appreciate having actual users helping develop this plugin, as you may be aware our resources are limited.

> Let's start by reading this https://github.com/spyder-ide/spyder/wiki/Dev%3A-Github-Workflow



I've read the link you sent, and I'd be happy to go ahead with that workflow.

**Comment by JamesOwers** [comment]
> Ok so I read the Pweave thread.
> 
> Option 1: Take over pweave
> Option 2: Try codebraid.
> 
> So just to be sure, does codebraid do the things needed? OR, is it "better"to update pweave @JamesOwers ?

Difficult to say at the moment before trying codebraid - though owner is completely unresponsive

Happy to crack on regardless - `pweave` still works, and we should write this plugin to support multiple backends eventually. Point me to an issue and I'll crack on with the fix when I can.

I can commit a couple of hours a week I think.  This workflow is how I work at work 😄 https://github.com/spyder-ide/spyder/wiki/Dev%3A-Github-Workflow

**Comment by goanpeca** [comment]
These are great news @JamesOwers. I need a couple of days to finish some pending work. Then I can clean this repo up to move to github actions, update to Spyder 4.x and start working on any enhancements :-) 

Great to have you on board and welcome!

**Comment by goanpeca** [comment]
We could  schedule a videocall to go over some of Spyder internal and give you a quick overview of the plans moving forward.  Let me know what you think.

**Comment by JamesOwers** [comment]
Sounds great. Currently trying to coordinate my travel back to the UK given pandemic (currently in US). Apologies for ongoing slow responses. Let me know how and when to connect and I shall.

**Comment by goanpeca** [comment]
Hi @JamesOwers, how are you doing? Back to UK ?

Well we can set an appointment. See https://calendly.com/goanpeca so you can pick some time that I have available. 

Cheers!

**Comment by JamesOwers** [comment]
Hullo @goanpeca - I am! Unfortunately, as you can imagine given new remote working, things have ramped up at work, and I'm much more time constrained. Realistically I'm not going to get more time till I finish mid July. Apologies for this situation!

**Comment by goanpeca** [comment]
Sure thing! Ping me when you want to tackle this one

Cheers!

**Comment by amichuda** [comment]
Hi! I just wanted to see if there's been any progress on this at all! I know that life is very different now so priorities might have changed, but just wanted to check in.

EDIT: I know that 6 months ago, I think the idea was either to take over maintaining `pweave` or using `codebraid` , but there's been some developments in that space with the very active  [`jupyter-book`](https://jupyterbook.org/intro.html) project, which might be another option.
